src/main/java/htsjdk/variant/variantcontext/writer/VariantContextWriterBuilder.java

```diff
@@ -445,7 +446,8 @@ else if (STREAM_TYPES.contains(this.outType))
                 if (options.contains(Options.INDEX_ON_THE_FLY))
                     throw new IllegalArgumentException("VCF index creation not supported for stream output.");
 
-                writer = createVCFWriter(null, outStream);
+                final OutputStream writerStream = outStream != null ? outStream : outStreamFromFile;
```

Comment:
If I'm reading this correctly in the case where you have VCF_STREAM as the type and outString != null we would have set outStreamFromFile to outStream on line 410. In which case we don't need writerStream at all and can just use
createVCFWriter(null, outStreamFromFile) on like 450